en/computer_vision/obstacle_avoidance.md

```diff
@@ -0,0 +1,166 @@
+
+
+## Mission Mode Avoidance {#mission_mode}
+
+PX4 supports obstacle avoidance in [Mission mode](../flight_modes/mission.md), using avoidance software running on a separate companion computer. 
+
+Obstacle avoidance is enabled within PX4 by [setting](../advanced_config/parameters.md) the [MPC_OBS_AVOID](../advanced_config/parameter_reference.md#MPC_OBS_AVOID) to 1.
+PX4 communicates with the obstacle avoidance software using an implementation of the MAVLink [Path Planning Protocol](https://mavlink.io/en/services/trajectory.html) (Trajectory Interface) which is [#described below](#mission_avoidance_interface).
+Provided an avoidance system complies with this interface it can be used with PX4.
+
+The tested companion computer platform is [Intel Aero](../flight_controller/intel_aero.md) running either the *local_planner* or *global_planner* avoidance software.
+This is set up as described in the [Intel Aero > Obstacle Avoidance](../flight_controller/intel_aero.md#obstacle-avoidance) and in the [PX4/avoidance](https://github.com/PX4/avoidance#obstacle-detection-and-avoidance) Github repo.
+
+
+### Mission Progression
+
+Mission behaviour with obstacle avoidance enabled is *slightly different* to the original plan.
+
+The difference when avoidance is active are:
+- A waypoint is "reached" when the vehicle is within the acceptance radius, regardless of its heading.
+  - This differs from normal missions, in which the vehicle must reach a waypoint with a certain heading (i.e. in a "close to" straight line from the previous waypoint). This constraint cannot be fulfilled when bstacle avoidance is active because the obstacle avoidance algorithm has full control of the vehicle heading, and the vehicle always moves in the current field of view. 
+- PX4 starts emitting a new current/next waypoint once the previous waypoint is reached (i.e. as soon as vehicle enters its acceptance radius).
+- If a waypoint is *inside* an obstacle it may unreachable (and the mission will be stuck). 
+  - If the vehicle projection on the line previous-current waypoint passes the current waypoint, the acceptance radius is enlarged such that the current waypoint is set as reached
+  - If the vehicle within the x-y acceptance radius, the altitude acceptance is modified such that the mission progresses (even if it is not in the altitude acceptance radius).
+- The original mission speed (as set in *QGroundControl*/PX4) is ignored.
+  The speed will be determined by the avoidance software:
+  - *local planner* mission speed is around 3 m/s.
+  - *global planner* mission speed is around 1-1.5 m/s.
+
+If PX4 stops receiving setpoint updates then obstacle avoidance will be disabled, and the mission continues under normal PX4 [Mission mode](../flight_modes/mission.md) control. 
```

Comment:
@mrivi FYI, this is where we would add "how long after obstacle avoidance messages stops that we got back to normal mission".

How does the system notify the user of this problem?

Reply:
0.5seconds. Right now there is no notification. Tanja is working on that right now.